src/components/ArticleDetails.jsx

```diff
@@ -0,0 +1,53 @@
+import { useParams, useNavigate } from "react-router";
+import { useState, useEffect } from "react";
+import { fetchArticleById } from "../api";
+
+export const ArticleDetails = () => {
+  const { articleId } = useParams();
+  const navigate = useNavigate();
+  const [article, setArticle] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setLoading(true);
+    setError(null);
+
+    fetchArticleById(articleId)
+      .then((articleData) => {
+        setArticle(articleData);
+        setLoading(false);
+      })
+      .catch(() => {
+        setError("Something went wrong while fetching the article.");
+        setLoading(false);
+      });
+  }, [articleId]);
+
+  if (loading) return <div>Loading article...</div>;
+  if (error) return <div>Error: {error}</div>;
+
+  return (
+    <article className="article-details">
+      <button onClick={() => navigate("/")} className="back-button">
```

Comment:
useNavigates convention is used for when you force a user to a page, for example, when you login and return to the home screen. You'd be better off using a Link here which is also better for accessibility, Links can be focused on using "tab" but a nav can't.